pyiron_atomistics/atomistics/structure/atoms.py

```diff
@@ -2561,22 +2561,71 @@ def get_initial_magnetic_moments(self):
             else:
                 return np.zeros(len(self))
 
-    def set_initial_magnetic_moments(self, magmoms=None):
+    def set_initial_magnetic_moments(self, magmoms):
         """
         Set array of initial magnetic moments.
 
         Args:
-            magmoms (numpy.ndarray/list): List of magneric moments
+            magmoms (numpy.ndarray/list/dict/float): List, dict or single value
+                of magneric moments
+
+
+        Example I:
+
+        >>> from pyiron_atomistics import Project
+        >>> structure = Project('.').create.structure.bulk('Ni', cubic=True)
+        >>> structure[-1] = 'Fe'
+        >>> v_dict = {'Fe': 1, 'Ni': 2}
+        >>> structure.set_initial_magnetic_moments(v_dict)
+        >>> print(structure.get_initial_magnetic_moments())
+
+        Output:
+
+        >>> array([2, 2, 2, 1])
+
+
+        Example II:
+
+        >>> from pyiron_atomistics import Project
+        >>> structure = Project('.').create.structure.bulk('Ni', cubic=True)
+        >>> structure[-1] = 'Fe'
+        >>> structure.set_initial_magnetic_moments(1)
+        >>> print(structure.get_initial_magnetic_moments())
+
+        Output:
+
+        >>> array([1, 1, 1, 1])
+
+        If you want to make it non-magnetic, set `None`:
+
+        >>> structure.set_initial_magnetic_moments(None)
+
+        If a list or an array of length 3 (e.g. [1, 2, 3]) is set and the
+        number of atoms is not 3, it will be considered as a non-collinear
+        magnetic moments, and the values will be distributed to all the
+        atoms (like in Example II)
         """
         # pyiron part
         if magmoms is not None:
+            if isinstance(magmoms, dict):
+                if set(self.get_chemical_symbols()) != set(magmoms.keys()):
+                    raise ValueError(
+                        "Elements in structure {} not equal to those in dict {}".format(
+                            set(self.get_chemical_symbols()), set(magmoms.keys())
+                        )
+                    )
+                magmoms = [magmoms[c] for c in self.get_chemical_symbols()]
+            elif not hasattr(magmoms, "__len__") or (
```

Comment:
```suggestion
            elif not isinstance(Sequence, magmoms) or (
```

plus `from collections.abc import Sequence`.

Reply:
Probably better to first handle the case that it's a sequence explicitely and then do this case rather than checking that it is not a sequence.

Reply:
> Probably better to first handle the case that it's a sequence explicitely and then do this case rather than checking that it is not a sequence.

I don't really understand what you mean by this.

Reply:
Apparently `ndarray` is not`Sequence`, so I cannot really use this one here.

Reply:
Maybe just `isinstance(magmoms, (ndarray, Sequence))` then.

Reply:
> > Probably better to first handle the case that it's a sequence explicitely and then do this case rather than checking that it is not a sequence.
> 
> I don't really understand what you mean by this.

Disregard, didn't read carefully enough.